src/tests/mod.rs

```diff
@@ -1,2 +1,22 @@
 mod conformance;
 mod resistance;
+
+/// Shorthand for `assert!(matches!(val, pattern), args..)`
+///
+/// Useful for asserting a enum matches a specific variant.
+///
+/// Usage:
+///     `assert_matches!(value, MyEnum::Variant(..));`
+///
+/// or with additional context:
+///     `assert_matches!(value, MyEnum::Variant(..), "Additional {}", "context");`
+#[macro_export]
+macro_rules! assert_matches {
```

Comment:
Lovely! Here's the [std implementation](https://doc.rust-lang.org/nightly/src/core/macros/mod.rs.html#140-165) and [tracking issue](https://github.com/rust-lang/rust/issues/82775) of this if you're curious.